### PR TITLE
fix(storageimage): handle onStorageGetError properly

### PR DESCRIPTION
**Pull request by dbanksdesign**
#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-storage/src/components/StorageImage/StorageImage.tsx [hunk 1]
```diff
@@ -23,7 +23,10 @@ export const StorageImage = ({
 
   const url = useStorageURL({
     key: imgKey,
-    options,
+    options: {
+      ...options,
+      validateObjectExistence: true,
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageImage/StorageImage.tsx:28]: Don't feel strongly but should `validateObjectExistence` be a prop with a default value of `true`?

Comment (dbanksdesign) [reply packages/react-storage/src/components/StorageImage/StorageImage.tsx]: Done

File: packages/react-storage/src/components/StorageImage/StorageImage.tsx [hunk 2]
```diff
@@ -14,11 +14,12 @@ export const StorageImage = ({
   identityId,
   imgKey,
   onStorageGetError,
+  validateObjectExistence = true,
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageImage/StorageImage.tsx:17]: What if the provided value is `false` or `null` 😅

Comment (dbanksdesign) [reply packages/react-storage/src/components/StorageImage/StorageImage.tsx:17]: It works? I just tried it. Or do you mean should we change this boolean prop to be the opposite so the default is false?

Comment (reesscot) [reply packages/react-storage/src/components/StorageImage/StorageImage.tsx:17]: Yeah, if they provide `false` or `null`, I think it would default to `true` right?

**Review by reesscot — APPROVED** [review]
LGTM, thanks for fixing this

**Comment by reesscot** [comment]
Missing changeset for new behavior

File: packages/react-storage/src/components/StorageImage/StorageImage.tsx (hunk 2 above)
Comment (calebpollman) [reply packages/react-storage/src/components/StorageImage/StorageImage.tsx:17]: > Yeah, if they provide false or null, I think it would default to true right

This 🤠 Maybe just add something like:
```
const resolvedValidateObjectExistence = isUndefined(validateObjectExistence) ? true : validateObjectExistence;
```

Comment (dbanksdesign) [reply packages/react-storage/src/components/StorageImage/StorageImage.tsx:17]: ugh fine. a bit wordy, but changed

Comment (calebpollman) [reply packages/react-storage/src/components/StorageImage/StorageImage.tsx]: Mean you could've renamed it 😏

**Review by calebpollman — DISMISSED** [review]

**Review by reesscot — APPROVED** [review]

**Review by reesscot — APPROVED** [review]
LGTM

**Review by calebpollman — APPROVED** [review]

**Review by reesscot — APPROVED** [review]
